doc/help/kdb-record.md

```diff
@@ -0,0 +1,144 @@
+# kdb-record(1) -- Session recording for Elektra
+
+## SYNOPSIS
+
+`kdb record <COMMAND>`
+
+Where `<COMMAND>` is one of the following:
+
+- `start [session_name]`: start or resume a recording session.
+- `stop`: stop the active recording session.
+- `reset [session_name]`: remove all logged data from the recording session.
+- `delete [session_name]`: delete the recording session.
+- `mark [--meta meta_name] <key_name>`: marks a key to be used for validation.
+- `list`: list all available recording sessions.
+- `name`: prints the name of the currently active session.
+- `changes [session_name]`: shows a human-readable list of changes.
+- `export  [--replay|--diff] [session_name]`: creates a file with the changes in the session.
+- `import [--disable-validation] <file_name>`: imports files created by the `export` command.
+
+## DESCRIPTION
+
+The `kdb record` utility is used to manage session recordings.
+Only a single recording session can be active per user at a time.
+A user can have multiple non-active sessions.
+
+If a session is active, every change to the keys, values and meta-data stored in Elektra will be logged into that session.
+
+## ENVIRONMENT VARIABLES
+
+- `ELEKTRA_RECORD_ACTIVE_SESSION`: If this environment variable is set, the specified session will be the active session.
```

Comment:
First of all 

> We could consider to use SUDO_USER (preferred to USER) to be used in our resolver. 

I'm fine with this. This would fix a lot of problems, especially with storing the recording session within Elektra itself. I will create an issue for this. If we do it this way, the `ELEKTRA_RECORD_ACTIVE_SESSION` will be dropped.


> It is unclear if [session_name] or ELEKTRA_RECORD_ACTIVE_SESSION will be used.

`ELEKTRA_RECORD_ACTIVE_SESSION` would be the active session.
If you specify the `[session_name]` argument for any command, you instruct that command NOT to use the active session. So that way `[session_name]` will be used.

> With the env var you don't have any control over how many sessions are open? 

Yes, this is a limitation of this approach. If we store the session in external files we don't have control over sessions that are controlled via `ELEKTRA_RECORD_ACTIVE_SESSION`. This is a trade-off. If we store them in Elektra like I initially proposed, we do actually have control over those sessions.

> It does not solve the sudo problem as sudo per default removes env vars.

Yes. But it can be instructed to keep them using the `-E` flag. This is something we should note in the manual.

Reply:
> It does not solve the sudo problem as sudo per default removes env vars.

True, but there is `sudo -E` to preserve the environment. So at least it is possible to solve it this way everything else that was suggested before didn't solve the sudo problem at all.

> We could consider to use `SUDO_USER` (preferred to USER) to be used in our resolver. 

I did not think of that. I admit, using `$SUDO_USER` in the `resolver` is probably a better solution than the env-var for the sudo problem. If we go down this root, we should definitely create a separate issue, since there is more to discuss here (e.g. permissions when creating files in `user:/` with `sudo`).

> With the env var you don't have any control over how many sessions are open? 

I assume you mean different shell sessions could use different `kdb record` sessions. IMO this would actually be an advantage, since you can't accidentally influence your recording from a different shell (or maybe from an active background process).

The currently  man page says "Only a single recording session can be active per user at a time.", but from a technical POV I think this can be seen as when a `kdbSet` happens exactly 0 or exactly 1 sessions are active.

> I think it is either/or: having a start/stop interface or using an env var.

Yes, probably just one would be better. Personally, I tend towards the env-var, but if we can solve all the problems, I'm fine with storing the active session in the KDB as well.